nexus_constructor/component_tree_model.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python
+from PySide2.QtCore import QAbstractItemModel, QModelIndex, Qt
+import PySide2.QtGui
+from PySide2.QtGui import QVector3D
+from nexus_constructor.component import ComponentModel
+from nexus_constructor.transformations import TransformationModel, TransformationsList
+import re
+
+
+def get_duplication_name(prototype_name, list_of_nodes):
```

Comment:
This is duplicating functionality that we already have in `ui_utils.generate_unique_name`

Reply:
Oh! Fixed.